src/components/forms/SignUpForm/index.tsx

```diff
@@ -1,39 +1,30 @@
-import React, { useState } from 'react';
-import { useForm } from 'react-hook-form';
+import React from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
 import i18next from 'i18next';
-import { useMutation } from 'react-query';
 
 import { User } from 'utils/types';
 import { emailRegex, passwordRegex } from 'constants/regex';
-import { signUp } from 'services/UsersService';
-import Loading from 'components/Spinner/components/loading';
 
-function SignUpForm() {
+type SignUpFormProps = {
```

Comment:
It's rare the case that we'll need to use `type` instead of `interface` since it's a bit more restrictive
```suggestion
interface SignUpFormProps {
```

Reply:
In my previous project, we had a lot of completely unused interfaces. So, we only generated interfaces when we needed much more functionality for typing. In this case, it seems to me that there is not much need to use an interface.
Let me know if it makes sense to you.